src/main/java/net/glowstone/block/itemtype/ItemProjectile.java

```diff
@@ -0,0 +1,20 @@
+package net.glowstone.block.itemtype;
+
+import net.glowstone.entity.GlowPlayer;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Projectile;
+import org.bukkit.inventory.ItemStack;
+
+/**
+ * An item that can be used to throw a projectile (egg, snowball, etc.)
+ */
+public abstract class ItemProjectile extends ItemType {
+
+    protected final EntityType entityType;
+
+    public ItemProjectile(EntityType entityType) {
+        this.entityType = entityType;
+    }
+
+    public abstract Projectile use(GlowPlayer player, ItemStack holding);
```

Comment:
Needs documentation

Reply:
Documented in a314f6a8d42113df74766f2043f9dbc9bb23a91d